src/Monolog/Handler/BrowserConsoleHandler.php

```diff
@@ -131,21 +138,27 @@ protected static function getResponseFormat(): string
         // Check content type
         foreach (headers_list() as $header) {
             if (stripos($header, 'content-type:') === 0) {
-                // This handler only works with HTML and javascript outputs
-                // text/javascript is obsolete in favour of application/javascript, but still used
-                if (stripos($header, 'application/javascript') !== false || stripos($header, 'text/javascript') !== false) {
-                    return 'js';
-                }
-                if (stripos($header, 'text/html') === false) {
-                    return 'unknown';
-                }
-                break;
+                return static::getResponseFormatFromContentType($header);
             }
         }
 
         return 'html';
     }
 
+    protected static function getResponseFormatFromContentType(string $contentType): ?string
+    {
+        // This handler only works with HTML and javascript outputs
+        // text/javascript is obsolete in favour of application/javascript, but still used
+        if (stripos($contentType, 'application/javascript') !== false || stripos($contentType, 'text/javascript') !== false) {
+            return static::FORMAT_JS;
```

Comment:
why not `self::` ?

Reply:
https://github.com/Seldaek/monolog/pull/1160

Reply:
Yeah on the other hand you used self:: below at line 166. I don't particularly think it matters as nobody should change/override these constants, but it'd be good to at least be consistent about it :)